### PR TITLE
Add library & example app

### PR DESCRIPTION
**Pull request by lukasmatta**
Project contains library and an example app to demonstrate its functionality.

Future work:
- add readme instructions
- add schematics to make setup easier for library consumers
- investigate creation of [custom angular cli builder](https://angular.io/guide/cli-builder) that would generate `categories.json` file, so that it wouldn't be necessary to run separate process for this purpose
- setup deployment of the example app to github pages
- investigate support for SSG to improve performance
- get rid of bootstrap dependency

### CODE REVIEW

File: projects/cps-mdoc-viewer/README.md [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+After building your library with `ng build cps-mdoc-viewer`, go to the dist folder `cd dist/cps-mdoc-viewer` and run `npm publish`.
+
+## Running unit tests
+
+Run `ng test cps-mdoc-viewer` to execute the unit tests via [Karma](https://karma-runner.github.io).
+
+## Further help
+
+To get more help on the Angular CLI use `ng help` or go check out the [Angular CLI Overview and Command Reference](https://angular.io/cli) page.
```
Comment (fateeand) [inline projects/cps-mdoc-viewer/README.md:25]:
```suggestion
# CPS MDoc Viewer

Angular library utilized in constructing websites that host tutorials and documentation, providing reference materials to help users learn about a particular topic, tool, or technology. The content is supplied in the format of markdown documents.

### License

Apache License 2.0 (see the [LICENSE](https://github.com/AbsaOSS/cps-mdoc-viewer/blob/master/LICENSE) file for the full text)
```

File: projects/cps-mdoc-viewer/package.json [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+  "peerDependencies": {
+    "@angular/common": "^17.3.0",
+    "@angular/core": "^17.3.0",
+    "cps-ui-kit": "^17.26.0",
+    "gray-matter": "^4.0.3",
+    "js-yaml": "^4.1.0",
+    "marked": "^12.0.2",
+    "ngx-markdown": "^17.2.1"
+  },
+  "dependencies": {
```
Comment (fateeand) [inline projects/cps-mdoc-viewer/package.json:13]: To be added
 
 "repository": {
    "type": "git",
    "url": "https://github.com/AbsaOSS/cps-mdoc-viewer"
  },
  "homepage": "https://github.com/AbsaOSS/cps-mdoc-viewer",
  "license": "Apache License 2.0",

File: projects/example-app/src/app/app.config.ts [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+ * limitations under the License.
+ */
+
+import { ApplicationConfig } from '@angular/core';
+import { provideCPSMDocViewer } from 'cps-mdoc-viewer';
+
+export const appConfig: ApplicationConfig = {
+  providers: [
+    provideCPSMDocViewer({
+      headerTitle: 'Example App',
```
[the remark this answers is not in the document]
Comment (lukasmatta) [reply projects/example-app/src/app/app.config.ts]: Addressed

File: projects/cps-mdoc-viewer/README.md (hunk 1 above)
Comment (lukasmatta) [reply projects/cps-mdoc-viewer/README.md]: Addressed

File: projects/cps-mdoc-viewer/package.json (hunk 2 above)
Comment (lukasmatta) [reply projects/cps-mdoc-viewer/package.json:13]: Added

**Review by fateeand — APPROVED** [review]

**Pull request merged**